src/SmartFormat/Utilities/PluralRules.cs

```diff
@@ -217,44 +231,44 @@ public static class PluralRules
         if (pluralWordsCount == 2) return value < 2 ? 0 : 1;
 
         if (pluralWordsCount == 3) return GetWordsCount3Value(value);
-
+        
         if (pluralWordsCount == 4) return GetWordsCount4Value(value);
 
         return -1;
     }; // DualFromZeroToTwo: one (n == 0..2 fractionate and n != 2), other
 
     private static int GetWordsCount3Value(decimal n)
     {
-        return n switch {
-            0 => 0,
+        return Math.Abs(n) switch
+        {
+            <=0 => 0,
             > 0 and < 2 => 1,
-            > 2 => 2,
-            _ => -1
+            >= 2 => 2
         };
     }
 
     private static int GetWordsCount4Value(decimal n)
     {
-        return n switch {
-            < 0 => 0,
-            0 => 1,
-            > 0 and < 2 => 2,
-            > 2 => 3,
-            _ => -1
+        return Math.Abs(n) switch
+        {
+            <=0 => 0,
+            > 0 and < 2 => 1,
+            >= 2 and < 3 => 2,
+            > 2 => 3
```

Comment:
Should this be `>= 3` instead of `> 2`? or maybe use the default to be safe?
`_ => 3`?

Reply:
👍